File: SfListView_VB/Form1.vb
Imports System.Collections.ObjectModel

Public Class Form1

    Dim countryInfoCollection As ObservableCollection(Of CountryInfo) = New ObservableCollection(Of CountryInfo)()
    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load


        SfListView1.DataSource = GetDataSource()
        SfListView1.ShowCheckBoxes = True

        SfListView1.DisplayMember = "CountryName"

    End Sub

    Public Function GetDataSource() As ObservableCollection(Of CountryInfo)

        countryInfoCollection.Add(New CountryInfo() With {
        .CountryName = "China",
        .Continent = "Asia"
    })
        countryInfoCollection.Add(New CountryInfo() With {
        .CountryName = "India",
        .Continent = "Asia"
    })
        countryInfoCollection.Add(New CountryInfo() With {
        .CountryName = "Japan",
        .Continent = "Asia"
    })
        countryInfoCollection.Add(New CountryInfo() With {
        .CountryName = "Malaysia",
        .Continent = "Asia"
    })
        countryInfoCollection.Add(New CountryInfo() With {
        .CountryName = "UnitedSates",
        .Continent = "NorthAmerica"
    })
        countryInfoCollection.Add(New CountryInfo() With {
        .CountryName = "Canada",
        .Continent = "NorthAmerica"
    })
        countryInfoCollection.Add(New CountryInfo() With {
        .CountryName = "Germany",
        .Continent = "Europe"
    })
        countryInfoCollection.Add(New CountryInfo() With {
        .CountryName = "UnitedKingdom",
        .Continent = "Europe"
    })
        countryInfoCollection.Add(New CountryInfo() With {
        .CountryName = "Italy",
        .Continent = "Europe"
    })
        countryInfoCollection.Add(New CountryInfo() With {
        .CountryName = "NewZealand",
        .Continent = "PacificIsland"
    })
        countryInfoCollection.Add(New CountryInfo() With {
        .CountryName = "Australia",
        .Continent = "PacificIsland"
    })
        countryInfoCollection.Add(New CountryInfo() With {
        .CountryName = "Finland",
        .Continent = "Europe"
    })
        countryInfoCollection.Add(New CountryInfo() With {
        .CountryName = "Sweden",
        .Continent = "Europe"
    })
        countryInfoCollection.Add(New CountryInfo() With {
        .CountryName = "Netherland",
        .Continent = "Asia"
    })
        Return countryInfoCollection
    End Function

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        For Each item In countryInfoCollection
            If item.CountryName = "Canada" OrElse item.CountryName = "Germany" OrElse item.Continent = "Europe" Then
                SfListView1.CheckedItems.Add(item)
            End If
        Next
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        For Each item In countryInfoCollection
            If item.Continent = "Europe" Then
                SfListView1.CheckedItems.Remove(item)
            End If
        Next
    End Sub
End Class

Public Class CountryInfo
    Public Property CountryName As String
    Public Property Continent As String
End Class

File: SfListView_C#/Form1.cs
using Syncfusion.DataSource;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace SfListVIewCheckBox
{
    public partial class Form1 : Form
    {
        ObservableCollection<CountryInfo> countryInfoCollection = new ObservableCollection<CountryInfo>();
        ObservableCollection<int> checkedIndexes = new ObservableCollection<int>();

        public Form1()
        {
            InitializeComponent();
            sfListView1.ShowCheckBoxes = true;
            sfListView1.DataSource = GetDataSource();
            sfListView1.ItemChecked += SfListView1_ItemChecked;

            sfListView1.CheckBoxSelectionMode = Syncfusion.WinForms.ListView.Enums.CheckBoxSelectionMode.CheckOnItemClick;
            sfListView1.DisplayMember = "CountryName";

            ObservableCollection<CountryInfo> GetDataSource()
            {
                countryInfoCollection.Add(new CountryInfo() { CountryName = "China", Continent = "Asia" });
                countryInfoCollection.Add(new CountryInfo() { CountryName = "India", Continent = "Asia" });
                countryInfoCollection.Add(new CountryInfo() { CountryName = "Japan", Continent = "Asia" });
                countryInfoCollection.Add(new CountryInfo() { CountryName = "Malaysia", Continent = "Asia" });
                countryInfoCollection.Add(new CountryInfo() { CountryName = "UnitedSates", Continent = "Asia" });
                countryInfoCollection.Add(new CountryInfo() { CountryName = "Canada", Continent = "Asia" });
                countryInfoCollection.Add(new CountryInfo() { CountryName = "Germany", Continent = "Asia" });
                countryInfoCollection.Add(new CountryInfo() { CountryName = "UnitedKingdom", Continent = "Asia" });
                countryInfoCollection.Add(new CountryInfo() { CountryName = "Italy", Continent = "Asia" });
                countryInfoCollection.Add(new CountryInfo() { CountryName = "NewZealand", Continent = "Asia" });
                countryInfoCollection.Add(new CountryInfo() { CountryName = "Australia", Continent = "Asia" });
                countryInfoCollection.Add(new CountryInfo() { CountryName = "Finland", Continent = "Asia" });
                countryInfoCollection.Add(new CountryInfo() { CountryName = "Sweden", Continent = "Asia" });
                countryInfoCollection.Add(new CountryInfo() { CountryName = "Netherland", Continent = "Asia" });

                return countryInfoCollection;
            }
            
        }

        private void SfListView1_ItemChecked(object sender, Syncfusion.WinForms.ListView.Events.ItemCheckedEventArgs e)
        {
            checkedIndexes.Clear();
            foreach (var allItems in sfListView1.View.Items)
            {
                foreach (var checkedItem in sfListView1.CheckedItems)
                {
                    if (allItems == checkedItem)
                    {
                        checkedIndexes.Add(sfListView1.View.DisplayItems.IndexOf(allItems));
                    }
                }
            }
            sfListView2.DataSource = checkedIndexes;
        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }
    }

    public class CountryInfo
    {
        public string CountryName { get; set; }
        public string Continent { get; set; }
    }
}
